gadget/ondisk.go

```diff
@@ -54,9 +60,16 @@ type sfdiskPartition struct {
 	// says --part-attrs takes a space or comma separated list, the output from
 	// --json/--dump uses a different format.
 	Attrs string `json:"attrs"`
-	Type  string `json:"type"`
-	UUID  string `json:"uuid"`
-	Name  string `json:"name"`
+	// Type is the structure type, which is the same as VolumeStructure's Type,
+	// see that doc-comment for full details, but note that sfdisk may format
+	// the type differently than would normally be used in gadget.yaml so
+	// conversion should be done before using this Type field directly.
+	Type string `json:"type"`
+	// UUID is the partition UUID for the partition.
```

Comment:
nitpick: this is self-evident, no comment is needed. OTOH, a comment on the Start and Size fields, saying whether we are talking sectors or bytes would be most welcome :-)

Reply:
Sure, I will take this in a followup as well